ecgtools/parsers.py

```diff
@@ -92,3 +100,252 @@ def cmip6_default_parser(
+                    # check to make sure that there's a 'glob_string' key for each ensemble entry
+                    if 'glob_string' not in stream_info.keys():
+                        print("ERROR: Each ensemble must contain a 'glob_string' key.")
+                        return False
+        return True
+
+    def _validate_yaml(self):
+        """
+        Validates the generic yaml input against the schema.  It uses either yamale or the internal validater.
+
+        Parameters
+        ----------
+        None
+
+        Returns
+        -------
+        boolean
+            True - passes the validation, False - fails the validation
+        """
+
+        # verify the format is correct
+        if self.validater == 'yamale':
+            try:
+                import yamale
+
+                print('Validating yaml file with yamale.')
+                cwd = Path(os.path.dirname(__file__))
+                schema_path = str(cwd.parent / 'schema') + '/generic_schema.yaml'
+                schema = yamale.make_schema(schema_path)
+                data = yamale.make_data(self.yaml_path)
+                try:
+                    yamale.validate(schema, data)
+                    print('Validation success! 👍')
+                    return True
+                except ValueError as e:
+                    print(
+                        'Yamale found that your file, '
+                        + self.yaml_path
+                        + ' is not formatted correctly.'
+                    )
+                    print(e)
+                    return False
+            except ImportError:
+                print('Validating yaml file internally.')
+                return self._internal_validation()
+        else:
+            print('Validating yaml file internally.')
+            return self._internal_validation()
+
+    def _parser_netcdf(self, filepath, local_attrs):
+        """
+        Opens a netcdf file in order to gather time and requested attribute information.
+        Also attaches assigned attributes gathered from the yaml file.
+
+        Parameters
+        ----------
+        filepath : str
+            The full path to the netcdf file to attatch attributes to.
+        local_attrs : dict
+            Holds attributes that need to be attached to the filenpath.
+
+        Returns
+        -------
+        dict
+            Returns all of the attributes that need to be assigned to the netcdf.
+        """
+
+        fileparts = {}
+        fileparts['path'] = filepath
+
+        try:
+            fileparts['variable'] = []
+            # open file
+            d = nc.Dataset(filepath, 'r')
+            # find what the time (unlimited) dimension is
+            dims = list(dict(d.dimensions).keys())
+            if 'time' in d.variables.keys():
+                times = d['time']
+                start = str(times[0])
+                end = str(times[-1])
+                fileparts['time_range'] = start + '-' + end
```

Comment:
Are `start` and `end` numeric values or string representation of corresponding datetime objects?

Reply:
But `times` is an array of `float`s, right?

How is the `fileparts['time_range']` attribute used (or supposed to be used)?  I can't find code where it is used.

Reply:
@kmpaul , that is correct.  `times` is a list of `float` values.

You are correct that it is not used within the code.  Instead `fileparts['time_range']` is inserted as a value within the data frame (or catalog) for that particular file.

Reply:
Ok.  Thanks for that @sherimickelson.

It sounds to me, then, like we want the `time_range` attribute to be "human readable," namely as a datetime string.  And maybe we want to split `time_range` into `start_time` and `end_time` separately?  Then each one can be queried separately?

@andersy005 What are your thoughts on this?

Reply:
Having `time_range` split into `start_time`, `end_time` will provide a better experience, and will play nicely with the search API once we add support for datetime objects in intake-esm.

Reply:
@kmpaul I went ahead and made this change